gear/gear/database.py

```diff
@@ -95,17 +95,19 @@ def __init__(self):
                 await cursor.execute('START TRANSACTION;')
 
     async def _aexit(self, exc_type, exc_val, exc_tb):
-        if not self.conn:
-            return
-
-        if exc_type:
-            await self.conn.rollback()
-        else:
-            await self.conn.commit()
-        self.conn = None
-
-        await aexit(self.conn_context_manager, exc_type, exc_val, exc_tb)
-        self.conn_context_manager = None
+        try:
+            if self.conn is not None:
+                conn = self.conn
+                # clear in case of exception
+                self.conn = None
```

Comment:
How does this work with `commit` having an assertion that self.conn can't be None.

Reply:
I'm calling conn.commit, not self.commit.  But either way I rewrote it slightly using a try ... finally block that is hopefully clearer.